examples/emulator.c

```diff
@@ -63,11 +64,12 @@ static gboolean on_socket_connected(GIOChannel *chan, GIOCondition cond,
 	if (fd == -1)
 		return FALSE;
 
-	/* Pick the first powered modem */
-	modem = modems->data;
+	/* Pick the last powered modem */
+	last = g_list_last(modems);
```

Comment:
Not really, powered_watch() and modem_watch() make sure there is a modem before this function is ever called. And also remove the watch when the modem list is empty. Also, there was no check in the previous code.